distribution/src/assembly/bin.xml

```diff
@@ -63,7 +63,7 @@
 
     <files>
         <file>
-            <source>target/WSO2Carbon/conf/default/deployment.yaml</source>
+            <source>carbon-home/conf/deployment.yaml</source>

```

Comment:
Yes, deployment.yaml is maintained at the product level since runtime is specific to the product. So in the bin.xml, we would need to copy this file per runtime conf